vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/TimelineItemFactory.kt

```diff
@@ -122,6 +124,10 @@ class TimelineItemFactory @Inject constructor(private val messageItemFactory: Me
                         Timber.v("Type ${event.root.getClearType()} not handled")
                         defaultItemFactory.create(params)
                     }
+                }.also {
+                    if (it != null && event.isEncrypted()) {
+                        decryptionFailureTracker.e2eEventDisplayedInTimeline(event.roomId, event.eventId, event.root.mCryptoError)
```

Comment:
Maybe just pass the TimelineEvent instead of 3 members of the TimelineEvent to simplify the signature of this method?